### ISSUE TITLE
geometry.conversions: rotation_matrix_to_axis_angle reports a reflection (det = -1) as a rotation; diag(-1, 1, 1) returns the identity

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4773]
`rotation_matrix_to_axis_angle` (and `rotation_matrix_to_quaternion` under it) accepts an improper matrix (`det = -1`) and returns a result that looks like a valid rotation. For the axis reflection `diag(-1, 1, 1)` the result is the zero vector, so the reflection is reported as "no rotation at all".

## Repro

```python
import torch
from kornia.geometry.conversions import rotation_matrix_to_axis_angle, rotation_matrix_to_quaternion, axis_angle_to_rotation_matrix

R = torch.diag(torch.tensor([-1.0, 1.0, 1.0]))  # a reflection, det = -1
print(torch.linalg.det(R).item())
print(rotation_matrix_to_axis_angle(R))
print(rotation_matrix_to_quaternion(R))
print(axis_angle_to_rotation_matrix(rotation_matrix_to_axis_angle(R)))
```

## Observed

```
-1.0
tensor([0., 0., 0.])
tensor([0.7071, 0.0000, 0.0000, 0.0000])
tensor([[1., 0., 0.],
        [0., 1., 0.],
        [0., 0., 1.]])
```

`diag(1, -1, 1)` and `diag(1, 1, -1)` also give `[0., 0., 0.]`. A general improper matrix (a rotation times `diag(-1, 1, 1)`) gives a finite vector that corresponds to neither the rotation nor the reflection. The quaternion step returns a non-unit `[0.7071, 0, 0, 0]`, and `quaternion_to_axis_angle` then maps it to the identity.

## Expected

A reflection has no axis-angle representation, so the call should not report it as a rotation. Any of these would do: raise, return NaN, or offer an opt-in check along the lines of the one proposed in #3961 for `camtoworld_to_worldtocam_Rt`.

## Why it is a bug

An axis flip that reaches this function, for example a graphics-to-vision frame change applied on one side only, goes through silently as the identity. Nothing tells the caller that the input was not a rotation. The docstring currently lists this case under "the input is **not** checked for being a rotation matrix". This issue asks for a decision on that behaviour. Related: #3961, where the same missing check makes a non-orthogonal `R` in `camtoworld_to_worldtocam_Rt` give a wrong inverse.

kornia `main` @ 57cd68e6f, torch 2.14.0, cpu.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by aminmiral** [comment]
I'd like to take this one.

Proposed fix, following the opt-in pattern merged in #4817 for `camtoworld_to_worldtocam_Rt`: add `check_rotation: bool = False` to `rotation_matrix_to_axis_angle` and `rotation_matrix_to_quaternion`, raising `ValueError` unless every input is a rotation — `max|R @ R^T - I|` within 100 × eps of the input dtype and `det(R) > 0`. The default path stays unchanged, so `diag(-1, 1, 1)` still returns `[0., 0., 0.]` unless the check is switched on.

That is one of the three options you list. Happy to make it raise by default, or return NaN, if you would rather decide the other way.

**Comment by ducha-aiki** [comment]
@aminmiral it's yours, and #4840 is the PR for it. We can't assign outside contributors, so this comment is the claim. Which of the three options the fix takes will be settled in the #4840 review. Whichever it is, the PR also updates the wart pin in `tests/geometry/test_conversions.py` and the #4773 note in `kornia/geometry/conversions.py`.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
